File: exercism/javascript/bob/bob.js
const { constant, cond, last } = require('lodash/fp')

const question = (q) => last(q) === '?'
const yell = (q) => q === q.toUpperCase()
const empty = (q) => q === ''
const otherwise = constant(true)

const cases = [
  [question, constant('Sure')],
  [yell, constant('Whoa, chill out!')],
  [empty, constant('Fine. Be that way!')],
  [otherwise, constant('Whatever.')]
]

class Bob {
  hey (q) {
    return cond(cases)(q)
  }
}

module.exports = Bob

File: exercism/haskell/bob/Bob.hs
module Bob (responseFor) where

import Data.Char

isYelling :: String -> Bool
isYelling q =
  (q == map toUpper q) && (q /= map toLower q)

isQuestion :: String -> Bool
isQuestion q =
  last q == '?'

isSilence :: String -> Bool
isSilence = all isSpace

responseFor :: String -> String
responseFor q
  | isSilence q = "Fine. Be that way!"
  | isYelling q = "Whoa, chill out!"
  | isQuestion q = "Sure."
  | otherwise = "Whatever."
